src/router_core/forwarder.c

```diff
@@ -154,7 +154,9 @@ qdr_delivery_t *qdr_forward_new_delivery_CT(qdr_core_t *core, qdr_delivery_t *in
     out_dlv->delivery_id = next_delivery_id();
     out_dlv->link_id     = out_link->identity;
     out_dlv->conn_id     = out_link->conn_id;
-    out_dlv->dispo_lock  = sys_mutex();
+    char name[64];
+    snprintf(name, sizeof(name), "delivery-%"PRIu32, out_dlv->delivery_id);
```

Comment:
This snprintf is in the fast path. Is this a problem ?

Reply:
It's probably overkill, especially for this lock which is hardly used.
In fact we probably want to avoid storing the mutex name (strdup call) when we're not explicitly debugging the locks.
Update coming soon...